src/Npgsql/TypeHandlers/CompositeHandlers/MappedCompositeHandler.cs

```diff
@@ -15,7 +15,7 @@ class MappedCompositeHandler<T> : NpgsqlTypeHandler<T>, IMappedCompositeHandler
         where T : new()
     {
         static readonly Func<T> Constructor = IsValueType<T>.Value
-            ? null
+            ? null!
```

Comment:
`System.FlatLieException()` 😄
No, honestly. I'd rather make this `Func<T>?` and put the bang in the one place where `Constructor` is actually used.
I agree that these kinds of things are necessary in lazy initialization scenarios, but if we start using it like this it defeats the whole purpose of nullable reference types IMO.
If this is a performance optimization it probably isn't worth it in the context of composite types.

Reply:
OK, no problem. Note that there is no perf impact as this is only a compile-time warning suppression (as opposed to a question mark which would be a runtime check).